modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java

```diff
@@ -0,0 +1,171 @@
+                   fields.append(',').append(searchRequest.source().storedFields().get(i));
+                }
+                String storedFieldsParamName = remoteVersion.before(Version.V_5_0_0_alpha4) ? "fields" : "stored_fields";
+                params.put(storedFieldsParamName, fields.toString());
+            }
+            return params;
+        }
+
+        static HttpEntity entity(BytesReference query) {
+            try (XContentBuilder entity = JsonXContent.contentBuilder();
+                    XContentParser queryParser = XContentHelper.createParser(query)) {
+                entity.startObject();
+                entity.field("query");
+                /*
+                 * We're intentionally a bit paranoid here - copying the query as xcontent rather than writing a raw field. We don't want
+                 * poorly written queries to escape. Ever.
+                 */
+                entity.copyCurrentStructure(queryParser);
+                XContentParser.Token shouldBeEof = queryParser.nextToken();
+                if (shouldBeEof != null) {
+                    throw new RuntimeException(
+                            "query was more than a single object. This first token after the object is [" + shouldBeEof + "]");
+                }
+                entity.endObject();
+                BytesReference bytes = entity.bytes();
+                return new ByteArrayEntity(bytes.array(), bytes.arrayOffset(), bytes.length(), ContentType.APPLICATION_JSON);
+            } catch (IOException e) {
+                throw new RuntimeException("Unexpected error building entity", e);
+            }
+        }
+
+        private static void addIndexesOrTypes(StringBuilder path, String name, String[] array) {
+            if (array == null || array.length == 0) {
+                return;
+            }
+            path.append(checkIndexOrType(name, array[0]));
+            for (int i = 1; i < array.length; i++) {
+                path.append(',').append(checkIndexOrType(name, array[i]));
+            }
+            path.append('/');
+        }
+
+        private static String checkIndexOrType(String name, String indexOrType) {
+            if (indexOrType.indexOf(',') >= 0) {
+                throw new IllegalArgumentException(name + " containing [,] not supported but got [" + indexOrType + "]");
+            }
+            if (indexOrType.indexOf('/') >= 0) {
+                throw new IllegalArgumentException(name + " containing [/] not supported but got [" + indexOrType + "]");
+            }
+            return indexOrType;
+        }
+
+        private static String sortToUri(SortBuilder<?> sort) {
+            if (sort instanceof FieldSortBuilder) {
+                FieldSortBuilder f = (FieldSortBuilder) sort;
+                return f.getFieldName() + ":" + f.order();
+            }
+            throw new IllegalArgumentException("Unsupported sort [" + sort + "]");
+        }
+    }
+
+    static class Scroll {
```

Comment:
This class seems... extraneous. It doesn't encapsulate any information, can the static methods be added elsewhere and this class removed?